src/components/ProductCreateEditPage/ProductCreateEditPage.tsx

```diff
@@ -0,0 +1,207 @@
+import { useTitle } from 'hooks/useTitle';
+
+import { ContentBox } from 'components/ContentBox/ContentBox';
+import { PageLayout } from 'components/PageLayout/PageLayout';
+import { ServiceContainerCreatingUpdating } from 'components/ServiceContainers/ServiceContainerCreatingUpdating';
+
+import * as productApi from 'services/productApi';
+
+import { createSafePatch } from 'utils/patchHelper';
+import { generatePageTitle } from 'utils/titleHelper';
+
+const formConfig = {
+  name: {
+    isRequired: true,
+  },
+  abbreviation: {
+    isRequired: true,
+  },
+};
+
+interface IProductCreateEditPageProps {
+  isEditPage?: boolean;
+}
+
+export const ProductCreateEditPage = ({ isEditPage = false }: IProductCreateEditPageProps) => {
+  const { productId } = useParams();
+  const navigate = useNavigate();
+
+  // create page
+  const serviceContainerCreatePage = useServiceContainer(productApi.createProduct);
+
+  // edit page - get method
+  const serviceContainerEditPageGet = useServiceContainer(productApi.getProduct);
+  const serviceContainerEditPageGetRunner = serviceContainerEditPageGet.run;
+
+  // edit page - patch method
+  const serviceContainerEditPagePatch = useServiceContainer(productApi.patchProduct);
+
+  const { register, setFieldValues, getFieldState, getFieldErrors, handleSubmit, isSubmitDisabled } = useForm();
+
+  useTitle(
+    generatePageTitle({
+      pageType: isEditPage ? 'Edit' : 'Create',
+      serviceContainer: serviceContainerEditPageGet,
+      firstLevelEntity: 'Product',
+    })
+  );
+
+  const submitCreate = (data: IFieldValues) => {
+    return serviceContainerCreatePage
+      .run({
+        serviceData: { data },
+      })
+      .then((response: any) => {
+        const newProductId = response?.data?.id;
+        if (!newProductId) {
+          throw new Error(`Invalid productId coming from Orch POST response: ${newProductId}`);
+        }
+        navigate(`/products/${newProductId}`);
+      })
+      .catch((e: any) => {
+        throw new Error('Failed to create Product.');
+      });
+  };
+
+  const submitUpdate = (data: IFieldValues) => {
+    const patchData = createSafePatch(serviceContainerEditPageGet.data, data);
+
+    serviceContainerEditPagePatch
+      .run({ serviceData: { id: productId, patchData } })
+      .then(() => {
+        navigate(`/products/${productId}`);
+      })
+      .catch(() => {
+        throw new Error('Failed to edit Product.');
+      });
+  };
+
+  useEffect(() => {
+    if (isEditPage) {
+      serviceContainerEditPageGetRunner({ serviceData: { id: productId } }).then((response: any) => {
+        setFieldValues(response.data);
+      });
+    }
+  }, [isEditPage, productId, serviceContainerEditPageGetRunner, setFieldValues]);
+
+  const formComponent = (
+    <ContentBox padding isResponsive>
+      <Form
+        onSubmit={(e) => {
+          e.preventDefault();
+        }}
+      >
+        <FormGroup
+          isRequired
```

Comment:
[optional] required information needs to be passed on 3 different places:
1) formConfig
2) FormGroup
3) Input

Is it possible to propagate it from formConfig to other places automatically?

Reply:
Won't do, useForm hook will not return props specific to PatternFly components.

Reply:
I don't feel it PatternFly specific, you could say exactly the same about other properties that are returned at this moment. Even if we use own input component or other 3rd party component, some `isRequired` flag would be still needed to prevent duplications. 

`useForm` is managing validations -> `isRequired` is directly related to validations, so also from this point of view, it's part of its scope.

The thing is that at this moment it's easy to create some inconsistency, if any of the `isRequired` is missing by mistake or there can be some unwanted left overs.

Reply:
We agreed it won't be implemented now, this decision can be revalidated again later.